src/main/java/ch/njol/skript/classes/ClassInfo.java

```diff
@@ -336,7 +368,14 @@ public Pattern[] getUserInputPatterns() {
 	public Changer<? super T> getChanger() {
 		return changer;
 	}
-	
+
+	@Nullable
+	public Supplier<T[]> getSupplier() {
+		if (supplier == null && c.isEnum())
+			supplier = c::getEnumConstants;
+		return supplier != null ? supplier : null;
```

Comment:
No need to check for null if you're returning s null and not using any of supplier's methods
```suggestion
		return supplier;
```